File: R_Trees_Java/Rectangle.java
package main;

public class Rectangle {
	private int coord_X;
	private int coord_Y;
	private int width;
	private int height;

	public Rectangle(int x, int y, int w, int h) {
		this.coord_X = x;
		this.coord_Y = y;
		this.width = w;
		this.height = h;
	}

	public int getX() {
		return coord_X;
	}

	public int getY() {
		return coord_Y;
	}

	public int getWidth() {
		return width;
	}

	public int getHeight() {
		return height;
	}

}

File: R_Trees_CPP/Rectangle.cpp
#include "Rectangle.h"

Rectangle::Rectangle(int x, int y, int w, int h, std::string id)
{
    this->posX = x;
    this->posY = y;
    this->width = w;
    this->height = h;
    this->id = id;
}

Rectangle::~Rectangle()
{
}

int Rectangle::getX()
{
    return this->posX;
}

int Rectangle::getY()
{
    return this->posY;
}

int Rectangle::getHeight()
{
    return this->height;
}

int Rectangle::getWidth()
{
    return this->width;
}

std::string Rectangle::getId()
{
    return this->id;
}

Node * Rectangle::getHijo()
{
    return this->hijo;
}


